### ISSUE TITLE
bittorrent deployment issue with version 4.4.0

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#196]
**Reported by efocht on 6 Jul 2006 07:39 UTC**
I'm trying the OSCAR5 trunk on fc5-x86_64 and the client isn't able to get the
boel_binaries from the master.

On the master the /var/log/systemimager/bittorrent-tracker.log is full of
messages like below:

```
Traceback (most recent call last):
  File "/usr/lib/python2.4/site-packages/BitTorrent/RawServer_twisted.py", line 418, in _make_wrapped_call
    function(*args)
  File "/usr/lib/python2.4/site-packages/BitTorrent/HTTPHandler.py", line 187, in data_came_in
    if not c.data_came_in(data) and not c.closed:
  File "/usr/lib/python2.4/site-packages/BitTorrent/HTTPHandler.py", line 52, in data_came_in
    self.next_func = self.next_func(val)
  File "/usr/lib/python2.4/site-packages/BitTorrent/HTTPHandler.py", line 96, in read_header
    r = self.handler.getfunc(self, self.path, self.headers)
  File "/usr/lib/python2.4/site-packages/BitTorrent/track.py", line 735, in get
    rsize = self.add_data(infohash, event, ip, paramslist)
  File "/usr/lib/python2.4/site-packages/BitTorrent/track.py", line 563, in add_data
    NatCheck(self.connectback_result,infohash,myid,ip1,port,self.rawserver)
  File "/usr/lib/python2.4/site-packages/BitTorrent/NatCheck.py", line 33, in __init__
    rawserver.async_start_connection((ip, port), self)
  File "/usr/lib/python2.4/site-packages/BitTorrent/RawServer_twisted.py", line 724, in async_start_connection
    if (len(self.pending_sockets) >= self.config['max_incomplete'] and
KeyError: 'max_incomplete'
```

Any idea what could be wrong?

What I see running on the master is: bittorrent-tracker and
launchmany-console. Torrent files are there, too.

**Comment by dikim33** [comment]
**Comment by bli on 6 Jul 2006 07:45 UTC**
BitTorrent 4.4.0 seems to have some issues with python-twisted present and enabled.  Therefore the fix (in SystemImager) is to disable it permanantly by calling ```bittorrent-tracker``` with the ```--twisted 0``` argument.

This was checked into SystemImager SVN trunk !r3660.

**Comment by dikim33** [comment]
**Comment by bli on 20 Sep 2006 07:19 UTC**
(In [5363]) This commit is a merge from trunk -> branch-5-0:
   svn merge -r5361:5362 https://svn.oscar.openclustergroup.org/svn/oscar/trunk .

Closes #195, #197, #239, #242 for milestone OSCAR 5.0.

Original commit message:
Update SystemImager to 3.7.4r3747 (soon to be released!) and adapted other scripts with API changes.
Please report any issues to oscar-devel@lists.sourceforge.net!
